src/librustc/mir/interpret/allocation.rs

```diff
@@ -16,15 +16,20 @@ use std::borrow::Cow;
 #[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable)]
 pub struct Allocation<Tag=(),Extra=()> {
     /// The actual bytes of the allocation.
-    /// Note that the bytes of a pointer represent the offset of the pointer
-    pub bytes: Vec<u8>,
+    /// Note that the bytes of a pointer represent the offset of the pointer. Bytes of undefined
+    /// allocations may be allocated lazily as bytes are initialized but either none or all bytes
+    /// are allocated.
+    pub(crate) bytes: Vec<u8>,
```

Comment:
```suggestion
    crate bytes: Vec<u8>,
```

Reply:
Requires a customized implementation in `src/librustc/ich/impls_ty.rs` where destructuring assignment could no longer capture all fields. Would that be fine?

Reply:
`crate` and `pub(crate)` are equivalent. Why would anything about that impl change?

Reply:
Oh. Right. My bad, I'm just not used to `crate field` syntax.